src/video-player.js

```diff
@@ -61,16 +62,38 @@ class VideoPlayer extends Utils.mixin(Eventable) {
     super();
 
     this.nbCalls = 0;
-
     this.videoElement = getResolveVideoElement(elem);
 
-    this.options = extractOptions(this.videoElement, initOptions);
-
-    this._videojsOptions = this.options.videojsOptions;
-
     // Make sure to add 'video-js' class before creating videojs instance
     this.videoElement.classList.add('video-js');
 
+    // first phase - shouldn't be exposed to people until we have BE ready
+    const profileName = initOptions.__profileName;
+
+    if (profileName) {
+      this.initProfile(profileName, initOptions)
+        .then((profileData) => this.initPlayer(profileData.playerOptions, ready))
+        .catch((e) => this.throwInvalidProfileError(e));
+    } else {
+      this.initPlayer(initOptions, ready);
+    }
+  }
+
+  initPlayer(initOptions, ready) {
+    // create empty profile options for simple extends
+    if (!this.profileOptions) {
+      this.profileOptions = {
+        playerOptions: {},
+        sourceOptions: {},
+        playlistOptions: {},
+        playlistByTagOptions: {},
+        sourcesByTagOptions: {}
+      };
```

Comment:
I'd prefer if we wouldn't need to do that. preparing empty placeholders for everything even if most won't need it is just adding noise.